### PR TITLE
Feature/list permissions

### PR DESCRIPTION
**Pull request by iamquan**

### CODE REVIEW

File: app/modules/security/_share/config/module.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {IModule, Module, MenuItem} from "../../../../common/models/layout";
+import {Permissions} from "./../../permission/permissions";
+import {AuthenticationMode} from "../../../../common/enum";
+let security: IModule = createModule();
+export default security;
+function createModule() {
+    let module = new Module("app/modules/security", "security");
+    module.menus.push(
+        new MenuItem(
+            "Security", "/Permissions", "fa fa-edit",
```
Comment (techcoaching) [inline app/modules/security/_share/config/module.ts]: Please get this from resource file if possible

File: app/modules/security/_share/config/module.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            new MenuItem("Permissions", "/Permissions", "")
+        )
+    );
+    module.addRoutes([
+        { path: "/permissions", name: "Permissions", component: Permissions, data: { authentication: AuthenticationMode.Require } },
```
Comment (techcoaching) [inline app/modules/security/_share/config/module.ts]: what does this mean, can we ignore this:
data: { authentication: AuthenticationMode.Require }

File: app/modules/security/_share/services/permissionService.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import configHelper from "../../../../common/helpers/configHelper";
+let permissionService = {
+    getPermissions: getPermissions
+};
+export default permissionService;
+function getPermissions() {
+    var url = String.format("{0}/permissions", configHelper.getAppConfig().api.baseUrl);
+    var connector = window.ioc.resolve("IConnector");
```
Comment (techcoaching) [inline app/modules/security/_share/services/permissionService.ts]: i think we can improve this, i i got the mistake if i type window.ioc.resolve("Connector");
please use constant for this

File: app/modules/security/permission/permissionsModel.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import {EventManager} from "../../../common/eventManager";
+export class PermissionsModel {
+    public actions: Array<any> = [];
+    public eventKey = "permissions_ondataloaded";
+    public options: any = {
+        data: [],
+        columns: [
+            { field: "name", title: "Name", index: 0 },
```
Comment (techcoaching) [inline app/modules/security/permission/permissionsModel.ts]: please get the text for title from resource file (i18n)

File: app/modules/usermanagement/_share/config/module.ts [hunk 5]
```diff
@@ -11,14 +11,14 @@ function createModule() {
         new MenuItem(
             "User Management", "/Roles", "fa fa-users",
```
Comment (techcoaching) [inline app/modules/usermanagement/_share/config/module.ts]: please move these texts to resource files (i18n): "User Management", the same comment for others

File: app/modules/security/_share/config/module.ts (hunk 2 above)
Comment (iamquan) [inline app/modules/security/_share/config/module.ts]: @techcoaching this module will force user to login if they haven't done it

File: app/modules/security/_share/services/permissionService.ts (hunk 3 above)
Comment (iamquan) [inline app/modules/security/_share/services/permissionService.ts]: @techcoaching I imported ```import {IoCType} from "../../../../common/enum";``` and call it ```var connector = IoCType.IConnector;``` but get error on ```return connector.get(url);``` property get doesn't exists on type IoCType

**Pull request closed without merging**